doc/specs/#1564 - Settings UI/spec.md

```diff
@@ -0,0 +1,102 @@
+```
+
+If people still like the UI but want to access the JSON file, we can provide an "Open the JSON file" button at the bottom of the navigation menu.
+
+### Top-level navigation
+
+The settings UI should have top-level navigation that aligns with the overall structure of the settings.json file. The following are the proposed navigation items:
+
+- Globals
+- Profiles
+- Color schemes
+- Key bindings
+
+### Launch methods
+
+#### Proposal 1 - Launch in a new window
+
+Clicking the settings button in the dropdown menu will open the settings UI in a new window. This allows the user to edit their settings and see the Terminal live update with their changes.
+
+In the Windows taskbar, the icon will appear as if Terminal has multiple windows open.
+
+#### Proposal 2 - Launch in a new tab
+
+Clicking the settings button in the dropdown menu will open the settings UI in a new tab. This helps us take steps toward supporting non-terminal content in a tab. However, tab tear-off should be implemented before we have this. This way, users can still see their changes take effect if they want by tearing out the settings UI.
+
+### Editing and saving settings
+
+#### Proposal 1 - Automatically save settings
+
+As users edit fields in the settings UI, they are automatically saved and written to the JSON file. This allows the user to see their settings changes taking place in real time.
+
+#### Proposal 2 - Implement a save button
+
+Users will only see their settings changes take place once they click "Save". Clicking "Save" will write to the settings.json file. This aligns with the functionality that exists today by editing the settings.json file in a text editor and saving it.
+
+A future option could be adding a TerminalControl inside the settings UI to preview what the changes will look like before actually saving them to the settings.json file.
+
+## UI/UX Design
+
+The purpose of the below image is to depict the navigation menu. The layout of the content in the page will be discussed in a future design review.
+
+![Settings UI navigation](./navigation.png)
+
+## Capabilities
+
+### Accessibility
+
+This will have to undergo full accessibility testing because it is a new UI element. All items inside the settings UI should be accessible by a screen reader and the keyboard.
+
+### Security
+
+This does not impact security.
+
+### Reliability
+
+This will not improve reliability.
+
+### Compatibility
+
+This will change the default experience to open the UI, rather than the JSON file in a text editor. This behavior can be reverted with the setting listed [above](#ability-to-disable-displaying-the-settings-ui).
+
+### Performance, Power, and Efficiency
+
+This does not affect performance, power, nor efficiency.
+
+## Potential Issues
+
+## Future considerations
+
+- We will have to have design reviews for all of the content pages.
+- We should have undo functionality. In a text editor, you can type `Ctrl+Z` however the settings UI is a bit more complex.

```

Comment:
I think going with a save and reset button does this?

Reply:
Yes, can have a link button with "Reset Defaults" but should also have <kbd>Ctrl+Z</kbd> option

Reply:
@cinnamon-msft So this is different from the Save/Reset buttons?